tools/cdk-release/lib/lifecycles/changelog.ts

```diff
@@ -18,25 +19,66 @@ export interface ChangelogOptions {
   verbose?: boolean;
   silent?: boolean;
 
+  changelogExperimentalChanges?: ExperimentalChanges;
   changeLogHeader?: string;
   includeDateInChangelog?: boolean;
   releaseCommitMessageFormat?: string;
 }
 
-export interface ChangelogResult {
-  readonly contents: string;
-  readonly changedFiles: string[];
-}
+export type ChangelogResult = { [changelogFile: string]: string };
 
-export async function changelog(
-  args: ChangelogOptions, currentVersion: string, newVersion: string, commits: ConventionalCommit[],
+export async function writeChangelogs(
+  args: ChangelogOptions,
+  currentVersion: Versions,
+  newVersion: Versions,
+  commits: ConventionalCommit[],
+  packages: PackageInfo[],
 ): Promise<ChangelogResult> {
+
   if (args.skip?.changelog) {
-    return {
-      contents: '',
-      changedFiles: [],
-    };
+    return {};
   }
+
+  const experimentalChangesTreatment = args.changelogExperimentalChanges ?? 'include';
+  const experimentalPackages = packages.filter(p => p.experimental);
+  const stableCommits = filterCommits(commits, { excludePackages: experimentalPackages.map(p => p.simplifiedName) });
+
+  if (experimentalChangesTreatment === 'include') {
+    const contents = await changelog(args, currentVersion.stableVersion, newVersion.stableVersion, commits);
+    return { [args.changelogFile]: contents };
+  } else if (experimentalChangesTreatment === 'strip') {
+    const contents = await changelog(args, currentVersion.stableVersion, newVersion.stableVersion, stableCommits);
+    return { [args.changelogFile]: contents };
+  } else if (experimentalChangesTreatment === 'separate') {
+    if (!currentVersion.alphaVersion || !newVersion.alphaVersion) {
+      throw new Error('unable to create "separate" changelogs without alpha package versions');
+    }
+
+    const changelogResults: ChangelogResult = {};
+
+    experimentalPackages.forEach(async (pkg) => {
+      const pkgCommits = filterCommits(commits, { includePackages: [pkg.simplifiedName] });
```

Comment:
Yup, that's a dual-edged sword; we could end up picking up completely unrelated changes because someone updated whitespace somewhere. Definitely thought about it, and decided to KISS for now.